Sources/ApolloSQLite/SQLiteNormalizedCache.swift

```diff
@@ -117,3 +117,14 @@ extension SQLiteNormalizedCache: NormalizedCache {
     try self.database.clearDatabase(shouldVacuumOnClear: self.shouldVacuumOnClear)
   }
 }
+
+private extension String {
+  func split(usingRegex pattern: String) -> [String] {
+    guard let regex = try? NSRegularExpression(pattern: pattern) else { return [] }
```

Comment:
Since `NSRegularExpression is an `NSObject`, in order to reduce the overhead incurred by initializing `NSObject`, we might be able to improve performance even just slightly more if we just make this:
```
/// A regex for splitting the cache key for a graphQLField into its path components.
///
/// Matches `.` so long as it's not contained within a parentheses (ie. being used in a field argument value).
static let fieldCacheKeyComponentRegex = NSRegularExpression(pattern: "\\.(?![^()]*\\))")

func splitIntoCacheKeyComponents() -> CacheKey {
...
}
```